### PR TITLE
[RNTester] fix getter of result from Image query cache

### PR DESCRIPTION
**Pull request by zhongwuzw**
## Summary

We assume `map` is the type of `Map`, but actually it's not, so we would get type error.

## Changelog

[iOS] [Fixed] - [RNTester] fix getter of result from Image query cache

## Test Plan

Before:
image

After:
We can get cache info.
image

### CODE REVIEW

File: React/Modules/RCTStatusBarManager.m [hunk 1]
```diff
@@ -108,8 +108,11 @@ - (void)applicationWillChangeStatusBarFrame:(NSNotification *)notification
     RCTLogError(@"RCTStatusBarManager module requires that the \
                 UIViewControllerBasedStatusBarAppearance key in the Info.plist is set to NO");
   } else {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```
Comment (zhongwuzw) [inline React/Modules/RCTStatusBarManager.m]: Remove warning.

File: RNTester/js/ImageExample.js [hunk 2]
```diff
@@ -93,7 +93,7 @@ class NetworkImageCallbackExample extends React.Component<
                     `✔ Prefetch OK (+${new Date() - mountTime}ms)`,
                   );
                   Image.queryCache([IMAGE_PREFETCH_URL]).then(map => {
-                    const result = map.get(IMAGE_PREFETCH_URL);
+                    const result = map[IMAGE_PREFETCH_URL];
```
[the remark this answers is not in the document]
Comment (cpojer) [reply RNTester/js/ImageExample.js]: @zhongwuzw seems like Flow isn't happy about this change?

File: React/Modules/RCTStatusBarManager.m (hunk 1 above)
Comment (cpojer) [reply React/Modules/RCTStatusBarManager.m]: Can you move this change into a separate PR?

Comment (zhongwuzw) [reply React/Modules/RCTStatusBarManager.m]: @cpojer Yes, please see #23625 .

**Comment by zhongwuzw** [comment]
@cpojer Hi, CI passes, the warning is because`queryCache`'s return type is not `Map`, that's also the reason this `PR` fixes.

**Comment by cpojer** [comment]
I'm assuming your changes to the Image files to `Promise<any>` are unintentional? Can you revert them?

**Comment by zhongwuzw** [comment]
@cpojer I'm intentional :joy:, because I found the return type is not `Map`, this is also cause of  `TypeError: map.get is not a function` error. Am I miss something?

**Comment by cpojer** [comment]
Ahhh.. now I see what you are trying to fix here. In that case, shouldn't the type be: `Promise<{string: 'memory' | 'disk' | 'disk/memory'}>`?

**Comment by zhongwuzw** [comment]
@cpojer The type is `Promise<{string: 'memory' | 'disk' | 'disk/memory'}>`.

**Pull request closed without merging**